aesara/tensor/math.py

```diff
@@ -622,11 +401,166 @@ def _c_all(self, node, name, inames, onames, sub):
         return decl, checks, alloc, loop, end
 
 
-class Max(NonZeroCAReduce):
-    nfunc_spec = ("max", 1, 1)
+class Max(COp):
```

Comment:
Why aren't we using the `NonZeroCAReduce` base class for this?

Reply:
@brandonwillard  Thank you for review. To fix this https://github.com/aesara-devs/aesara/pull/874#issuecomment-1118590762 problem with `Max` I coped the code from `MaxAndArgmax` and made it  `COp`. Do you think that it can add some problems?

Reply:
It could interfere with existing code that assumes `Max` is a subclass of `NonZeroCAReduce` (e.g. rewrites, our JAX and/or Numba implementations, etc.)

Also, we generally want to make use of existing code.  In this case, subclassing `NonZeroCAReduce` could make adding a C implementation much easier that it otherwise would be.